src/api/resolve-template.js

```diff
@@ -16,6 +18,28 @@ module.exports = function resolveTemplate(options, { supportedTemplates }) {
     path.basename(options.template || ''),
     options.libraryVersion
   );
+  let supportedVersion;
+
+  try {
+    const templateConfig = getAppTemplateConfig(
+      path.resolve('src/templates', templateName)
+    );
+    supportedVersion = templateConfig.supportedVersion;
+  } catch (error) {
+    // The template doesn't provide a configuration file `.template.js`.
```

Comment:
There are other errors that can happen.
Example getAppTemplateConfig throws an error when the template is there but is invalid.
Generally it's a bad idea to swallow errors, especially if you don't do anything with them afterwards

Reply:
Other errors are already checked before with `getAppTemplateConfig()` ([example here](https://github.com/algolia/create-instantsearch-app/blob/9e2216f06ff10f37a2a2aeb465d5ae0c1dc7646c/src/cli/getConfiguration.js)).

We swallow this error here because we also support custom templates (created by other users), which might not follow the exact same specification.

IMO this is a right use case of swallowing errors. What would you replace it with?

Note that if the error happens, this is not a blocker. This feature is a UX improvement to warn the user that the template is not supported by the selected version.

Reply:
What would be the downside of not catching errors here and letting the code throw them?